crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs

```diff
@@ -59,16 +63,23 @@ pub(crate) fn unnecessary_literal_union<'a>(checker: &mut Checker, expr: &'a Exp
 
     // Raise a violation if more than one
     if literal_exprs.len() > 1 {
-        let diagnostic = Diagnostic::new(
+        let literal_members: Vec<String> = literal_exprs
+            .into_iter()
+            .map(|expr| checker.locator().slice(expr.as_ref()).to_string())
+            .collect();
+
+        let mut diagnostic = Diagnostic::new(
             UnnecessaryLiteralUnion {
-                members: literal_exprs
-                    .into_iter()
-                    .map(|expr| checker.locator().slice(expr.as_ref()).to_string())
-                    .collect(),
+                members: literal_members.clone(),
             },
             expr.range(),
         );
 
+        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
+            format!("Literal[{}]", literal_members.clone().join(", ")),
```

Comment:
Can this `.clone()` be removed? We don't use `literal_memebrs` again, so it should be fine to consume it here.